backend/src/main/java/gov/cdc/usds/simplereport/utils/DateTimeUtils.java

```diff
@@ -37,6 +37,7 @@ public class DateTimeUtils {
   private static final ZoneId hawaiiTimeZoneId = ZoneId.of("US/Hawaii");
   private static final ZoneId aleutianTimeZoneId = ZoneId.of("US/Aleutian");
   private static final ZoneId samoaTimeZoneId = ZoneId.of("US/Samoa");
+  private static final ZoneId puertoRicoTimeZoneId = ZoneId.of("America/Puerto_Rico");
```

Comment:
In addition to specifically handling Puerto Rico, I think it makes sense to catch whatever exception was being thrown and default the timezone to something (eastern time?).  If this is a big deal we can add alerting for when it happens, but I think the failure state should be that the file still gets uploaded successfully.

Do we know how the single entry flow was handling this case?

Reply:
Re: handling the exception 

Based on the fix implemented I don't think there would be an exception thrown anymore.

Because we were doing this before

> return ZoneId.of("US/" + timezoneInfo.timezoneCommonName);

if `timezoneInfo.timezoneCommonName` returned something like `Atlantic` (like the valid Puerto Rico address did), we'd be trying to get the `ZoneId.of("US/Atlantic")` which would throw the exception.

Now, since we are using the `timezoneInfo.utcOffset` value instead of creating strings (that potentially don't exist) we should be good. If `timezoneInfo == null` we already return `null`. The method that calls ` getZoneIdByAddress` sets the default timezone (eastern time) if the value is `null`.

Reply:
Re: single entry flow
I believe this is [how we are handling it](https://github.com/CDCgov/prime-simplereport/blob/58ea0d7253b340647cc869ca82cdb07d3c2f15d6/backend/src/main/java/gov/cdc/usds/simplereport/api/converter/FhirConverter.java#L1334). We aren't extracting any timezone data from the ordering provider so we would not run into the same issue in the single entry flow based on my read of this 👀